polkadot/runtime/parachains/src/assigner_coretime/mod.rs

```diff
@@ -432,16 +440,11 @@ impl<T: Config> Pallet<T> {
 
 		// Update is needed:
 		let update = CoreSchedules::<T>::take((next_scheduled, core_idx));
-		let new_first = update.as_ref().and_then(|u| u.next_schedule);
-		descriptor.current_work = update.map(Into::into);
-
-		descriptor.queue = new_first.map(|new_first| {
-			QueueDescriptor {
-				first: new_first,
-				// `last` stays unaffected, if not empty:
-				last: queue.last,
-			}
-		});
+		if let Some(update) = update {
```

Comment:
map_or?

Reply:
I think we'd have to make `update_from_schedule` return the core descriptor for it to work? You think it's worth making the change? To me the current version is more readable, but I don't have a background in functional programming like you. 😄